authz/providers/azure/rbac/checkaccessreqhelper.go

```
@@ -371,6 +425,49 @@ func getAuthInfoListForWildcard(subRevReq *authzv1.SubjectAccessReviewSpec, stor
 	return authInfoList, nil
 }
 
+func getAuthInfoListForCustomResource(subRevReq *authzv1.SubjectAccessReviewSpec, clusterType string) ([]azureutils.AuthorizationActionInfo, error) {
+	var authInfoList []azureutils.AuthorizationActionInfo
+	if subRevReq.ResourceAttributes.Verb == "*" {
+		for _, action := range getCustomResourceOperationsMap(clusterType) {
+			authInfoList = append(authInfoList, action)
+		}
+	} else {
+		action := getActionName(subRevReq.ResourceAttributes.Verb)
+		authInfoSingle, found := getCustomResourceOperationsMap(clusterType)[action]
+		if !found {
+			return nil, errors.Errorf("No actions found for verb %s", subRevReq.ResourceAttributes.Verb)
+		}
+		authInfoList = append(authInfoList, authInfoSingle)
+	}
+
+	for i := range authInfoList {
+		err := setAuthInfoResourceAttributes(&authInfoList[i], subRevReq)
+		if err != nil {
+			return nil, errors.Errorf("Error while setting resource attributes: %s", err.Error())
+		}
+	}
+
+	return authInfoList, nil
+}
+
+func setAuthInfoResourceAttributes(action *azureutils.AuthorizationActionInfo, subRevReq *authzv1.SubjectAccessReviewSpec) error {
+	if subRevReq.ResourceAttributes == nil {
+		return errors.New("Resource attributes are empty")
+	}
+	if subRevReq.ResourceAttributes.Group == "" {
+		return errors.New("Group is empty")
+	}
```

I'd return an error if there is no version as well

---

@arxhive See my previous coment about version.